### PR TITLE
Cache player locales

### PR DESCRIPTION
**Pull request by kashike**
This changes `getLocale` to use a `LoadingCache` to get `Locale`s, instead of creating a new `Locale` each call.

### CODE REVIEW

**Comment by jamierocks** [comment]
:+1:

File: src/main/java/org/spongepowered/common/util/LanguageUtil.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.apache.commons.lang3.LocaleUtils;
+import org.spongepowered.api.util.annotation.NonnullByDefault;
+
+import java.util.Locale;
+
+@NonnullByDefault
+public final class LanguageUtil {
+
+    public static final LoadingCache<String, Locale> LOCALE_CACHE = CacheBuilder.newBuilder()
+        .build(new CacheLoader<String, Locale>() {
```
Comment (gratimax) [inline src/main/java/org/spongepowered/common/util/LanguageUtil.java]: Could just be `.build(LocaleUtils::toLocale)`?

Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/util/LanguageUtil.java]: @gratimax I believe `CacheLoader` is an abstract class, and method references don't work with them.

Comment (kashike) [inline src/main/java/org/spongepowered/common/util/LanguageUtil.java]: @JBYoshi is correct.

**Pull request closed without merging**